process/block/export_test.go

```diff
@@ -820,3 +827,48 @@ func GetLastExecutionResultsRootHash(
 func GetHaveTimeForProposal(startTime time.Time, maxDuration time.Duration) func() time.Duration {
 	return getHaveTimeForProposal(startTime, maxDuration)
 }
+
+func ConstructPartialShardBlockProcessorForTest(subcomponents map[string]interface{}) (*shardProcessor, error) {
+	bp := &baseProcessor{}
+	sp := &shardProcessor{baseProcessor: bp}
+
+	setField := func(target any, name string, component any) error {
+		rv := reflect.ValueOf(target).Elem()
+		field := rv.FieldByName(name)
+		if !field.IsValid() {
+			return fmt.Errorf("invalid field: %s", name)
+		}
+		if !field.CanSet() {
+			// bypass export check (ok in tests, same package)
+			field = reflect.NewAt(field.Type(), unsafe.Pointer(field.UnsafeAddr())).Elem()
+		}
+
+		val := reflect.ValueOf(component)
+		switch {
+		case val.Type().AssignableTo(field.Type()):
+			field.Set(val)
+		case val.Type().ConvertibleTo(field.Type()):
+			field.Set(val.Convert(field.Type()))
+		case val.Kind() != reflect.Ptr && field.Kind() == reflect.Ptr && val.Type().AssignableTo(field.Type().Elem()):
+			ptr := reflect.New(val.Type())
+			ptr.Elem().Set(val)
+			field.Set(ptr)
+		case val.Kind() != reflect.Ptr && field.Kind() == reflect.Ptr && val.Type().ConvertibleTo(field.Type().Elem()):
+			ptr := reflect.New(field.Type().Elem())
+			ptr.Elem().Set(val.Convert(field.Type().Elem()))
+			field.Set(ptr)
+		default:
+			return fmt.Errorf("cannot set field %s (got %s, expected %s)", name, val.Type(), field.Type())
+		}
+		return nil
+	}
+
+	for name, component := range subcomponents {
+		if err := setField(sp, name, component); err != nil {
+			if err2 := setField(sp.baseProcessor, name, component); err2 != nil {
+				return nil, err
```

Comment:
is it intended here err or should it be err2? (perhaps rename err2 to errSetBaseProcessor)